lib/Mojo/URL.pm

```diff
@@ -58,15 +58,13 @@ sub parse {
   my ($self, $url) = @_;
 
   # Official regex from RFC 3986
-  $url =~ m!^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\?([^#]*))?(#(.*))?!;
-  $self->scheme($2)                         if defined $2;
-  $self->path($5)                           if defined $5;
-  $self->query($7)                          if defined $7;
-  $self->fragment(_decode(url_unescape $9)) if defined $9;
-  if (defined(my $auth = $4)) {
-    $self->userinfo(_decode(url_unescape $1)) if $auth =~ s/^([^\@]+)\@//;
-    $self->host_port($auth);
-  }
+  $url =~ m!^(?:([^:/?#]+):)?(?://(?:([^@/]+?)\@)?([^/?#]*))?([^?#]*)(?:\?([^#]*))?(?:#(.*))?!;
+  $self->scheme($1)                         if defined $1;
+ 	$self->userinfo(_decode(url_unescape $2)) if defined $2;
+	$self->host_port($3);
```

Comment:
What's up with the borked indentation? Didn't you perltidy the file before opening the pull request?